test/skimOutputter.cpp

```diff
@@ -0,0 +1,1833 @@
+      bool SVfitConv_jetdownTot                      = tauH_SVFIT_mass_jetdownTot > 0;
+
+      if (doDNN)
+      {
+        DNNreader.SetShiftedInputs(bjet1_jetupTot, bjet2_jetupTot, tau1, tau2, vbfjet1_jetupTot, vbfjet2_jetupTot, met_jetupTot, svfit_jetupTot,
+            HHKin_mass_jetupTot, HHKin_chi2_jetupTot, KinFitConv_jetupTot, SVfitConv_jetupTot, MT2_jetupTot,
+            mTtot_jetupTot, pzeta_vis_jetupTot, pzeta_jetupTot, topMasses_jetupTot.first, topMasses_jetupTot.second, mt1_jetupTot, mt2_jetupTot);
+        std::vector<float> outs_jetupTot = DNNreader.GetPredictions();
+        DNNoutSM_kl_1_jetupTot = outs_jetupTot.at(0);
+
+        DNNreader.SetShiftedInputs(bjet1_jetdownTot, bjet2_jetdownTot, tau1, tau2, vbfjet1_jetdownTot, vbfjet2_jetdownTot, met_jetdownTot, svfit_jetdownTot,
+            HHKin_mass_jetdownTot, HHKin_chi2_jetdownTot, KinFitConv_jetdownTot, SVfitConv_jetdownTot, MT2_jetdownTot,
+            mTtot_jetdownTot, pzeta_vis_jetdownTot, pzeta_jetdownTot, topMasses_jetdownTot.first, topMasses_jetdownTot.second, mt1_jetdownTot, mt2_jetdownTot);
+        std::vector<float> outs_jetdownTot = DNNreader.GetPredictions();
+        DNNoutSM_kl_1_jetdownTot = outs_jetdownTot.at(0);
+      }
+
+      if (doBDT)
+      {
+        BDTreader.SetInputValues(bjet2_jetupTot.Pt(), (bjet1_jetupTot+bjet2_jetupTot).Pt(), tau1.Pt(),
+          tau2.Pt(), svfit_jetupTot.Pt(), BDT_channel,
+          BDT_HT20_jetupTot, pzeta_jetupTot, pzeta_vis_jetupTot, BDT_ditau_deltaPhi_jetupTot,
+          BDT_tauHsvfitMet_deltaPhi_jetupTot, mT_tauH_MET_jetupTot, mTtot_jetupTot, MT2_jetupTot,
+          BDT_MX_jetupTot, BDT_bH_tauH_MET_InvMass_jetupTot, BDT_bH_tauH_SVFIT_InvMass_jetupTot,
+          BDT_bH_tauH_InvMass_jetupTot, HHKin_mass_jetupTot, HHKin_chi2_jetupTot, BDT_MET_bH_cosTheta_jetupTot);
+        std::vector<float> BDTouts_jetupTot = BDTreader.GetPredictions();
+        BDToutSM_kl_1_jetupTot = BDTouts_jetupTot.at(0);
+
+        BDTreader.SetInputValues(bjet2_jetdownTot.Pt(), (bjet1_jetdownTot+bjet2_jetdownTot).Pt(), tau1.Pt(),
+          tau2.Pt(), svfit_jetdownTot.Pt(), BDT_channel,
+          BDT_HT20_jetdownTot, pzeta_jetdownTot, pzeta_vis_jetdownTot, BDT_ditau_deltaPhi_jetdownTot,
+          BDT_tauHsvfitMet_deltaPhi_jetdownTot, mT_tauH_MET_jetdownTot, mTtot, MT2_jetdownTot,
+          BDT_MX_jetdownTot, BDT_bH_tauH_MET_InvMass_jetdownTot, BDT_bH_tauH_SVFIT_InvMass_jetdownTot,
+          BDT_bH_tauH_InvMass_jetdownTot, HHKin_mass_jetdownTot, HHKin_chi2_jetdownTot, BDT_MET_bH_cosTheta_jetdownTot);
+        std::vector<float> BDTouts_jetdownTot = BDTreader.GetPredictions();
+        BDToutSM_kl_1_jetdownTot = BDTouts_jetdownTot.at(0);
+      }
+    }
+
+    // ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+    // Fill new branches
+    // Central values
+    b_HHKin_mass_new   ->Fill();
+    b_HHKin_chi2_new   ->Fill();
+    b_MT2_new          ->Fill();
```

Comment:
The central values are already saved in the original tree, why do we save them again? Unlike the central values of DNN(or BDT), who could change if the DNN training is updated, these are not supposed to change

Reply:
yeah, for now I just wanted to store them for comparison, later we can delete them.